common/src/service/IotHubConnectionString.cs

```diff
@@ -11,9 +11,11 @@
 
 namespace Microsoft.Azure.Devices
 {
-    internal sealed class IotHubConnectionString : IAuthorizationHeaderProvider, ICbsTokenProvider
+    /// <summary>
+    /// The properties required for authentication to IoT hub using a connection string.
+    /// </summary>
+    internal sealed class IotHubConnectionString : IotHubCredential
```

Comment:
This is not the best name for this class but I do not want to make this a daunting PR so I will make the naming changes in a different PR. Feel free to leave suggestions though.
This PR moves a part of the properties in this class that are common to auth to a base class called - IotHubCredential. Each of the auth types will derive from it and add implementation to get tokens.

Reply:
`IotHubConnectionStringCredential`?

Reply:
Sounds reasonable. Have conversation below with Tim on another class. We should look at it together to find good names. I will do one PR just to get the names right so that noone has to worry about logic. Just that so many names are already taken, I'm finding it hard to name these well :(

Reply:
I will send out another PR with these changes and see what everyone thinks.